src/Yesod/Paginator.hs

```diff
@@ -51,11 +51,19 @@ module Yesod.Paginator
 
     -- * Paginators
     , paginate
+    , paginateWith
     , selectPaginated
+    , selectPaginatedWith
 
     -- * Widgets
     , simple
+    , simpleWith
     , ellipsed
+    , ellipsedWith
+
+    -- * Pagination configuration
+    , PaginationConfig(..)
+    , PageParamName(..)
```

Comment:
Should these be exported like this or is it preferable if the config is constructed via the default config, e.g. `defaultPaginationConfig { paginationConfigPerPage = 5 }`?

Reply:
I'm on the fence. What you're describing, with only exporting the `default` function and setters, would be following [this pattern](https://www.snoyman.com/blog/2016/11/designing-apis-for-extensibility). It's nice, and its goals are reasonable, but I typically don't do things like that and tend to export simply. WDYT?

Reply:
Thanks for the link, that was indeed what I meant. I've seen it in a couple of packages, but I have no personal preference so I went for the simpler approach and just kept it as it is.